becquerel/tools/materials.py

```diff
@@ -1,202 +1,218 @@
-
-    Returns:
-      requests object.
-
-    Raises:
-      NISTMaterialsRequestError: if there was a problem making the request.
-
+    Returns
+    -------
+    materials
+        Dictionary keyed by material names containing the material data.
     """
-
-    req = requests.get(url)
-    if not req.ok or req.reason != "OK" or req.status_code != 200:
-        raise NISTMaterialsRequestError(
-            "NIST materials request failed: reason={}, status_code={}".format(
-                req.reason, req.status_code
-            )
-        )
-    return req
-
-
-def fetch_element_data():
-    """Retrieve data for the elements.
-
-    Data are found in Table 1:
-      http://physics.nist.gov/PhysRefData/XrayMassCoef/tab1.html
-
-    Returns:
-      A pandas DataFrame of the material data. Colums are 'Z', 'Symbol',
-      'Element', 'Z_over_A', 'I_eV', and 'Density'.
-
-    Raises:
-      NISTMaterialsRequestError: if there was a problem obtaining the data.
-
+    # fetch the data sources
+    data_elem = fetch_element_data()
+    data_mat = fetch_compound_data()
+    data_comp = fetch_compendium_data()
+
+    # perform various checks on the Compendium data
+    for j in range(len(data_comp)):
+        name = data_comp["Material"].values[j]
+        rho1 = data_comp["Density"].values[j]
+        rho2 = None
+        if name in data_elem["Element"].values:
+            rho2 = data_elem["Density"][data_elem["Element"] == name].values[0]
+        elif name in data_mat["Material"].values:
+            rho2 = data_mat["Density"][data_mat["Material"] == name].values[0]
+        if rho2:
+            if not np.isclose(rho1, rho2, atol=2e-2):
+                raise MaterialsError(
+                    f"Material {name} densities do not match between different data sources:  {rho1:.6f}  {rho2:.6f}"
+                )
+
+    for j in range(len(data_comp)):
+        name = data_comp["Material"].values[j]
+        if name in data_mat["Material"].values:
+            weight_fracs1 = data_comp["Composition_symbol"].values[j]
+            weight_fracs2 = data_mat["Composition_symbol"][
+                data_mat["Material"] == name
+            ].values[0]
+            if len(weight_fracs1) != len(weight_fracs2):
+                raise MaterialsError(
+                    f"Material {name} has different number of weight fractions in the different sources: {weight_fracs1}  {weight_fracs2}"
+                )
+            for k in range(len(weight_fracs1)):
+                elem1, frac1 = weight_fracs1[k].split(" ")
+                elem2, frac2 = weight_fracs2[k].split(" ")
+                if elem1 != elem2:
+                    raise MaterialsError(
+                        f"Material {name} weight fraction elements do not match between different data sources:  {elem1}  {elem2}"
+                    )
+                frac1 = float(frac1)
+                frac2 = float(frac2)
+            if not np.isclose(frac1, frac2, atol=3e-4):
```

Comment:
Is there a reason the `atol` are different for different properties?

Reply:
Yes, some of the data have changed in the Compendium versus the same materials in the NIST datasets. Those tolerances are just to show that the numbers are generally close but were found by increasing until the tests passed.